### PR TITLE
<compare>: Improve floating-point spaceship CPO codegen

### PR DESCRIPTION
**Pull request by statementreply**
Implements floating-point `strong_order` and `weak_order` without branching on the sign bit.

Also adds signaling NaN test cases, enabled on clang only (MSVC "quiets" signaling NaNs when constant evaluated).

### CODE REVIEW

File: stl/inc/compare [hunk 1]
```diff
@@ -396,35 +396,37 @@ namespace _Strong_order {
...
-                const bool _Right_negative = (_Right_uint & _Traits::_Shifted_sign_mask) != 0;
+                const _Uint_type _Left_shifted_sign  = _Left_uint & _Traits::_Shifted_sign_mask;
+                const _Uint_type _Right_shifted_sign = _Right_uint & _Traits::_Shifted_sign_mask;
 
-                // 3. Fast path: any negative value is less than any positive value.
-                if (_Left_negative != _Right_negative) {
-                    return _Left_negative ? strong_ordering::less : strong_ordering::greater;
-                }
+                // 3. Interpret floating-point bit patterns as sign magnitude representations of integers,
+                //    and then transform then into one's complement representation.
```
Comment (StephanTLavavej) [inline stl/inc/compare:414]:
```suggestion
                //    and then transform them into one's complement representation.
```

File: stl/inc/compare [hunk 2]
```diff
@@ -507,47 +510,42 @@ namespace _Weak_order {
...
-                _Right_uint &= ~_Traits::_Shifted_sign_mask;
+                if (_Right_magnitude > _Infinity_plus_one) {
+                    _Right_uint = _Right_shifted_sign | _Infinity_plus_one;
+                }
 
-                // 6. Fold all NaN values together.
-                // (The fact that _Infinity_plus_one represents a signaling NaN is irrelevant here.)
-                constexpr _Uint_type _Infinity_plus_one = _Traits::_Shifted_exponent_mask + 1;
+                // 4. Interpret floating-point bit patterns as sign magnitude representations of integers,
+                //    and then transform then into two's complement representation.
```
Comment (StephanTLavavej) [inline stl/inc/compare:533]:
```suggestion
                //    and then transform them into two's complement representation.
```

**Review by StephanTLavavej — COMMENTED** [review]
Thanks, this is a great improvement - I will push fixes for a couple of comment typos but everything else looks like crystalline perfection to me. :gem:

**Review by StephanTLavavej — APPROVED** [review]

**Comment by statementreply** [comment]
@StephanTLavavej I pushed fixes for comment typos after your approval.

**Review by cbezault — APPROVED** [review]
This is a pretty cool change. I got to learn a couple of ones' complement tricks today.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks for increasing the spaceship's speed - onward to Alpha Centauri! :rocket: :flying_saucer: :ringed_planet:

**Pull request merged**